tests/test_mysetting.rs

```diff
@@ -0,0 +1,207 @@
+use binrw::{io::Cursor, BinRead};
+use rekordcrate::setting::*;
+
+macro_rules! read_mysetting {
+    ($path:literal) => {{
+        let data = include_bytes!($path);
+        println!("Setting file: {}", $path);
+        let mut reader = Cursor::new(data);
+        let setting = Setting::read(&mut reader).expect("failed to parse setting file");
+        let result = match setting.data {
+            SettingData::MySetting(x) => Some(x),
+            _ => None,
+        };
+        result.expect("failed to match data section")
```

Comment:
```suggestion
        assert!(matches!(setting.data, SettingData::MySetting(_)), "failed to match data section") 
```

Reply:
This asserts that `setting.data` is of the expected type, but doesn't actually destructure. The macro still returns a variable of type `SettingData`, whereas the original version returns a `MySetting` struct.

Reply:
good point, I didn't consider the function in its outer context.